server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridTiler.java

```diff
@@ -147,8 +150,9 @@ private int setValuesForFullyContainedTile(String hash, GeoShapeCellValues value
         }
     }
 
-    class GeoTileGridTiler implements GeoGridTiler {
-        public static final GeoTileGridTiler INSTANCE = new GeoTileGridTiler();
+    class GeoTileGridTiler<G extends MultiGeoValues.GeoValue> implements GeoGridTiler<G> {
+        public static final GeoTileGridTiler INSTANCE = new GeoTileGridTiler<>();
+        public static final GeoTileGridTiler BOUNDED_INSTANCE = new GeoTileGridTiler<BoundedGeoShapeCellValues.BoundedGeoValue>();
```

Comment:
I guess I am missing something but I cannot tell the benefits of having two instances, they seem to be used the same way?